src/Symfony/Component/Serializer/Encoder/ChainDecoder.php

```diff
@@ -32,6 +32,8 @@ public function __construct(array $decoders = array())
 
     /**
      * {@inheritdoc}
+     * 
+     * @throws RuntimeException If no decoder is found.
```

Comment:
my case is not specific to DecoderInterface. It is true for any part of Symfony. I just took this one as example among your changes.

Documenting ``@throws`` in the interface is fine when the interface wants implementations to throw specific exceptions for some error cases. A plain ``@throws Exception`` means nothing.
And this does not forbid other exceptions to happen when something goes really wrong (would you then also document things like ``@throws \TypeError in case you don't respect the typehint`` to be complete ?). Thus, if your goal is that any possible exception throw by calling the method appears in the phpdoc, it means you have to account for all dependencies too, which gets unmaintainable very quickly (or you have to put try/catch everywhere to hide exceptions of your dependencies, even though they would appear only when your code is really buggy and so should be handled only by the global error handler displaying a 500 page and logging the error).